lib/src/init.dart

```diff
@@ -75,7 +75,8 @@ DynamicLibrary _openRealmLib() {
     try {
       return open();
     } on Error catch (e) {
-      ex ??= e; // remember first exception, if everything fails
+      ex ??= LoadRealmLibraryError();
```

Comment:
I think we should be really fixing this code instead. This may collect errors from the various places we try to load but then throw the error on why it fails the last open. The last open should be really succeeding if all previous do fail. 
Also this code has more issues. For example `_getNearestProjectRoot` is actually searching for `pubspec.yaml` file and without parsing it trying to load the library from some location near it. Instead we should be looking for the library we are trying to load directly.

Reply:
The issue related to this PR is fixed by building the library on ubuntu 20.4 in order it to be possible the library to be loaded on ubuntu 20 and 22.
This change here only prints all the errors instead of the wrong one. This is done in order to have more information in case we have similar issue later.
We can discuss and fix the code for loading library in another PR.

Reply:
Tested that both ubuntu 20 and 22 are loading the library.

Reply:
Discussed to be done with throwing a RealmError.